solution/dqt.api/QualifiedTeacherStatusService.cs

```diff
@@ -25,11 +27,17 @@ public QualifiedTeacherStatusService(IQualifiedTeachersService qtsService)
             ILogger log)
         {
             log.LogInformation("C# HTTP trigger function processed a request.");
-            string requestBody = await new StreamReader(req.Body).ReadToEndAsync();
 
-            dynamic data = JsonConvert.DeserializeObject(requestBody);
+            if (!req.Headers.ContainsKey(AUTH_HEADER)
+                || req.Headers[AUTH_HEADER][0] != Environment.GetEnvironmentVariable("APIKey"))
+            {
+                return new UnauthorizedResult();
+            }
+
+            string requestBody = await new StreamReader(req.Body).ReadToEndAsync();
+            var request = JsonConvert.DeserializeObject<ExistingQualifiedTeacherRequest>(requestBody);
+            var results = await _qtsService.GetQualifiedTeacherRecords(request.TeacherReferenceNumber, request.NINumber);
 
-            var results = await qtsService.GetQualifiedTeacherRecords("TRN1234", "NI1234");
             return new JsonResult(results);
         }
     }
```

Comment:
validate the request and return bad request for invalid request
add try catch block to handle internal server errors
return OK result with the response JSON object